tla/Traceccfraft.tla

```diff
@@ -345,7 +346,7 @@ TraceMatched ==
 
 TraceStateSpace ==
     \* TODO This can be removed when Traceccfraft is done.
-    /\ JsonFile \in KnownScenarios => TraceStats.distinct = Len(TraceLog)
+    /\ JsonFile \in KnownScenarios => TraceStats.distinct >= Len(TraceLog)
```

Comment:
@achamayou I will try to look into it this week.  Is it the case for all known scenarios, or just a subset of those?

Reply:
All of them grow by at least one, but sometimes more than one state. One of the scenarios still fails, I will look at why today.

Reply:
Changing from "equals" to "greater than or equals" causes `TraceMatched` to imply `TraceStateSpace`.   In other words, `TraceStateSpace` could be removed.

Reply:
The off-by-one violation of `TraceStateSpace` appears to be an undetected regression, likely caused by a change of `TraceInit`.  It seems as if the path to the generated trace logs in our CI integration is not `../build/`, making `TraceStateSpace` vacuously true.  In short, the off-by-one violation has nothing to do with this change.

However, validation of `bad_network` fails at diameter 101 once the off-by-one violation of `TraceStateSpace` has been corrected because `HandleRequestVoteResponse` is constraint to Candidates.

Reply:
https://github.com/microsoft/CCF/pull/5351

Reply:
@lemmy it is definitely "../build/" in the CI, as specified under https://github.com/microsoft/CCF/blob/main/.azure-pipelines-templates/trace_validation.yml#L11, but `KnownScenarios` was a subset of what we run.

I do not understand why we've changed the constraint to be `distinct - 1`, where it was `distinct` previously. That change applies to all files too. There's also not a lot of information in the trace error.